regEx/challenges-04.test.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 1
+//
+// Write a function named isNum that takes in a string or number of any length.
+// This function should use a regular expression pattern to return true if the input
+// contains a number, and false if the input does not contain a number.
+//
+// Example input/output:
+// 12345 => true
+// '12345' => true
+// 'h3llo world' => true
+// 'hello world' => false
+// ------------------------------------------------------------------------------------------------
+
+const isNum = (num) => {
+    let regex = /[0-9]/;
+    return regex.test(num);
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 2
+//
+// Write a function named isCapitalized that takes in a string. This function should use a
+// regular expression pattern to match all words that begin with a capital letter.
+//
+// Return an array containing all the matches.
+// ------------------------------------------------------------------------------------------------
+
+const isCapitalized = (string) => {
+  let regex = /^[A-Z]/;
+  const capitalArr = [];
+  let newStr = string.split(' ');
+  newStr.forEach( (word) => {
+    if(regex.test(word)){
+      capitalArr.push(word);
+    }
+  });
+  return capitalArr;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named citiesAtoJ that takes in an array of city names and uses a regular expression
+// pattern to return a new array containing any cities that begin with the letters A through J, inclusive.
+// ------------------------------------------------------------------------------------------------
+
+const citiesAtoJ = (cities) => {
+  let regex = /^[A-J]/;
+  const newArr = [];
+  cities.forEach( (city) => {
+    if(regex.test(city)){
+      newArr.push(city);
+    }
+  });
+  return newArr;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 4
+//
+// You have created a game application and begin by asking users an easy question: In which month is Halloween?
+//
+// Write a function named matchMonth which uses a regular expression pattern to match any of these inputs:
+// October, Oct, october, oct
+//
+// If the user enters any of these four inputs, return true. For ANY other input, return false.
+// Do not use the vertical bar (pipe) in your pattern.
+// ------------------------------------------------------------------------------------------------
+
+const matchMonth = (input) => {
+  let regex = /^([Oo]ct)(ober)?$/;
```

Comment:
you don't need the `()` around the `Oct` as that part is not optional. You would just have it around `ober`